crates/iceberg/src/test/create_test_table_s3.py

```diff
@@ -0,0 +1,199 @@
+        pa.field("name", pa.string(), nullable=True),  # StringType, required=False
+        pa.field("b", pa.bool_(), nullable=False),  # BooleanType, required=True
+        pa.field(
+            "ts", pa.timestamp("us"), nullable=False
+        ),  # TimestampType, required=True
+        pa.field("dt", pa.date32(), nullable=False),  # DateType, required=True
+        # pa.field("es", pa.struct([  # StructType, required=True
+        #     pa.field("a", pa.bool_(), nullable=False)  # NestedField(1, "a", BooleanType, required=True)
+        # ]), nullable=False),
+        # pa.field("m", pa.map_(  # MapType, required=True
+        #     pa.string(),  # key_type=StringType
+        #     pa.int64(),  # value_type=LongType
+        # ), nullable=False)
+    ]
+)
+
+partition_spec = PartitionSpec(
+    PartitionField(source_id=4, field_id=1000, transform=DayTransform(), name="date")
+)
+
+try:
+    print("Deleting existing table, if any")
+    catalog.drop_table("iceberg_test.test_table")
+except:
+    pass
+
+print("Creating Iceberg table")
+
+table = catalog.create_table(
+    "iceberg_test.test_table", schema, location=location, partition_spec=partition_spec
+)
+
+# Number of records
+num_records = args.rows
+
+print(f"Generating {num_records} rows")
+
+# Generate random data
+if args.json_file:
+    pandas_df = pd.read_json(args.json_file, lines=True)
+    pandas_df["ts"] = pd.to_datetime(pandas_df["ts"])
+    pandas_df["dt"] = pd.to_datetime(pandas_df["dt"]).dt.date
+else:
+    data = {
+        "id": np.arange(1, num_records + 1, dtype=np.int64),  # Sequential IDs
+        "name": np.random.choice(
+            [None, "Alice", "Bob", "Charlie", "Diana"], size=num_records
+        ),  # Random names or None
+        "b": np.random.choice([True, False], size=num_records),  # Random boolean values
+        "ts": [
+            datetime.datetime(2023, 1, 1) + datetime.timedelta(seconds=i)
+            for i in range(num_records)
+        ],  # Sequential timestamps
+        "dt": [
+            datetime.datetime(2023, 1, 1).date() + datetime.timedelta(days=i % 365)
+            for i in range(num_records)
+        ],  # Sequential dates (cycle annually)
+        # "es": [{"a": np.random.choice([True, False])} for _ in range(num_records)],  # Struct with random boolean
+        # "m": [{f"key_{j}": np.random.randint(1, 100) for j in range(np.random.randint(1, 5))} for _ in range(num_records)],  # Random maps
+    }
+    pandas_df = pd.DataFrame(data)
+
+print("Generating Pandas dataframe")
+
+# pyiceberg does not support nanosecond timestamps
+pandas_df["ts"] = pandas_df["ts"].astype("datetime64[us]")
+
+# print(pandas_df.head())
+
+print("Converting Pandas dataframe to Arrow")
+
+arrow_table = pa.Table.from_pandas(pandas_df, schema=arrow_schema)
+
+print("Writing data to Iceberg table")
+
+table.append(arrow_table)
+
+# Extract and print metadata location.
+print(table.inspect.metadata_log_entries().column("file")[-1].as_py())
```

Comment:
I found that it's useful to always put python code into a function, makes it easier to maintain.